Bottom: a23d8585160d52d515b3e735117844e06b92388d
Top:    17b765dc9232c1b293b1caecf8b7953d6ae42116
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-07-05 17:14:08 +0100

action.c: Fix handling of the case where a segment already exists when drawing lines

We cancel creating any segment which would overlay an existing, but we previously
failed to move the start-point and toggle the line-start direction so as to
continue drawing from the desired point as if we HAD created the line.


---

diff --git a/src/action.c b/src/action.c
index 8cd389c..b8b3fe7 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1255,8 +1255,11 @@ NotifyMode (void)
 	  if ((Crosshair.AttachedLine.Point1.X !=
 	       Crosshair.AttachedLine.Point2.X
 	       || Crosshair.AttachedLine.Point1.Y !=
-	       Crosshair.AttachedLine.Point2.Y)
-	      && (line =
+	       Crosshair.AttachedLine.Point2.Y))
+            {
+              PinType *via;
+
+              if ((line =
 		  CreateDrawnLineOnLayer (CURRENT,
 					  Crosshair.AttachedLine.Point1.X,
 					  Crosshair.AttachedLine.Point1.Y,
@@ -1265,12 +1268,12 @@ NotifyMode (void)
 					  Settings.LineThickness,
 					  2 * Settings.Keepaway,
 					  MakeFlags (line_flags))) != NULL)
-	    {
-	      PinType *via;
+                {
 
-	      addedLines++;
-	      AddObjectToCreateUndoList (LINE_TYPE, CURRENT, line, line);
-	      DrawLine (CURRENT, line);
+                  addedLines++;
+                  AddObjectToCreateUndoList (LINE_TYPE, CURRENT, line, line);
+                  DrawLine (CURRENT, line);
+                }
 	      /* place a via if vias are visible, the layer is
 	         in a new group since the last line and there
 	         isn't a pin already here */
@@ -1303,8 +1306,9 @@ NotifyMode (void)
 	    }
 	  if (PCB->Clipping && (Note.X != Crosshair.AttachedLine.Point2.X
 				|| Note.Y !=
-				Crosshair.AttachedLine.Point2.Y)
-	      && (line =
+				Crosshair.AttachedLine.Point2.Y))
+            {
+              if ((line =
 		  CreateDrawnLineOnLayer (CURRENT,
 					  Crosshair.AttachedLine.Point2.X,
 					  Crosshair.AttachedLine.Point2.Y,
@@ -1312,11 +1316,12 @@ NotifyMode (void)
 					  Settings.LineThickness,
 					  2 * Settings.Keepaway,
 					  MakeFlags (line_flags))) != NULL)
-	    {
-	      addedLines++;
-	      AddObjectToCreateUndoList (LINE_TYPE, CURRENT, line, line);
-	      IncrementUndoSerialNumber ();
-	      DrawLine (CURRENT, line);
+                {
+                  addedLines++;
+                  AddObjectToCreateUndoList (LINE_TYPE, CURRENT, line, line);
+                  IncrementUndoSerialNumber ();
+                  DrawLine (CURRENT, line);
+                }
 	      /* move to new start point */
 	      Crosshair.AttachedLine.Point1.X = Note.X;
 	      Crosshair.AttachedLine.Point1.Y = Note.Y;
